community/planning/capabilities-repository.md

```diff
@@ -0,0 +1,413 @@
+Saplings.
+
+### Capabilities
+
+A Capability is a set of dependencies that make up a broader application.  These
+dependencies include saplings and smart contracts.  They are akin to a
+application distribution.  For example, a Grid Product Capability would include
+the smart contracts required and the saplings that provide a web UI. Future
+parts of the stack included in a capability manifest may include state delta
+export and CLI WASM plugins.
+
+### Manifests and Packaging
+
+Each artifact in the repository will provide a manifest that describes its
+name, version, and required dependencies (be those file dependencies or other
+artifacts in the repository).  Manifests are specific to the artifact and
+optimized for their target environment.
+
+Likewise, artifacts will be packaged in a way that is optimized for their target
+environment.
+
+For example, a smart contract archive is a scar file.  This file is
+bzip2-compressed and contains a manifest and the WASM smart contract.  The
+manifest is a yaml file containing information about the smart contract, such as
+name and version.  The target environment for the scar file is the Sawtooth
+Sabre transaction handler via the Scabbard service.
+
+### Repository Layout
+
+The repository will be laid out in a static manner.  This choice is important,
+as it supports hosting the repositories on services like Amazon S3.  This layout
+will have at its root the major component divisions, with artifacts stored in a
+directory matching the artifact. This directory will store all versions of the
+artifact and a file with the SHA-512 hash of the artifact.
+
+Depending on the component, there may be additional resources that have been
+expanded out of the archive package.  For example, saplings may optionally
+include preview images.  These images will be available directly in the
+repository, so that client applications do not have to extract the entire
+archive in order to preview the potential saplings.
+
+The repository will include an index scheme that will provide mappings for
+things such names or dependency relationships.
+
+For example, the Grid Product, version 1.0, smart contract would be stored at
+
+`/contracts/grid_product/grid_product-1.0.scar`
+
+### Tools
+
+Interacting with the capabilities repository requires a set of tools.  For
+developers, these  for creating artifacts and their manifests, as well as
+publishing artifacts to the repository.  For repository maintainers, tools
+include removing artifacts and generating indexes files of the repository.
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Manifest Formats
+
+All the major components in the repository have a manifest describing their
+name, and version, as well as optional fields like description.
+
+#### Saplings
+The sapling manifest is a JSON file, as this is the most readily consumable
+format by the browsers
```

Comment:
s/by/for/, or "...the format most readily consumable by web browsers"

Reply:
Reworded as suggested